File: version1/EventLogHttpClientV1.go
package version1

import (
	"os"
	"reflect"
	"time"

	cdata "github.com/pip-services3-go/pip-services3-commons-go/data"
	clients "github.com/pip-services3-go/pip-services3-rpc-go/clients"
)

var (
	systemEventV1DataPageType = reflect.TypeOf(&SystemEventV1DataPage{})
	systemEventV1Type         = reflect.TypeOf(&SystemEventV1{})
)

type EventLogHttpClientV1 struct {
	*clients.CommandableHttpClient
}

func NewEventLogHttpClientV1() *EventLogHttpClientV1 {
	c := EventLogHttpClientV1{}
	c.CommandableHttpClient = clients.NewCommandableHttpClient("v1/eventlog")
	return &c
}

func (c *EventLogHttpClientV1) GetEvents(correlationId string, filter *cdata.FilterParams, paging *cdata.PagingParams) (*SystemEventV1DataPage, error) {
	params := cdata.NewEmptyStringValueMap()
	c.AddFilterParams(params, filter)
	c.AddPagingParams(params, paging)

	val, err := c.CallCommand(systemEventV1DataPageType, "get_events", correlationId, params, nil)
	if err != nil {
		return nil, err
	}
	page, _ := val.(*SystemEventV1DataPage)
	return page, nil
}

func (c *EventLogHttpClientV1) LogEvent(correlationId string, event *SystemEventV1) error {
	event.Time = time.Now()
	if event.Source == "" {
		event.Source, _ = os.Hostname()
	}

	params := cdata.NewAnyValueMapFromTuples(
		"event", event,
	)

	_, err := c.CallCommand(nil, "log_event", correlationId, nil, params.Value())
	return err
}

File: src/version1/EventLogHttpClientV1.ts
let os = require('os');

import { ConfigParams } from 'pip-services3-commons-node';
import { IReferences } from 'pip-services3-commons-node';
import { FilterParams } from 'pip-services3-commons-node';
import { PagingParams } from 'pip-services3-commons-node';
import { DataPage } from 'pip-services3-commons-node';
import { CommandableHttpClient } from 'pip-services3-rpc-node';

import { SystemEventV1 } from './SystemEventV1';
import { IEventLogClientV1 } from './IEventLogClientV1';

export class EventLogHttpClientV1 extends CommandableHttpClient implements IEventLogClientV1 {

    constructor(config?: any) {
        super('v1/eventlog');

        if (config != null)
            this.configure(ConfigParams.fromValue(config));
    }
        
    public getEvents(correlationId: string, filter: FilterParams, paging: PagingParams,
        callback: (err: any, page: DataPage<SystemEventV1>) => void) {
        this.callCommand(
            'get_events',
            correlationId,
            {
                filter: filter,
                paging: paging
            },
            callback
        );
    }

    public logEvent(correlationId: string, event: SystemEventV1,
        callback?: (err: any, event: SystemEventV1) => void) {

        event.time = event.time || new Date();
        event.source = event.source || os.hostname(); 

        this.callCommand(
            'log_event',
            correlationId,
            {
                event: event
            }, 
            callback
        );
    }

}
